diff --git a/taichi/common/core.h b/taichi/common/core.h
index f112f8f..e704943 100644
--- a/taichi/common/core.h
+++ b/taichi/common/core.h
@@ -167,38 +167,38 @@ using real = float32;
 
 // Float literal for both float32/64
 // (Learned from https://github.com/hi2p-perim/lightmetrica-v2)
-real constexpr operator"" _f(long double v) {
+real constexpr operator""_f(long double v) {
   return real(v);
 }
-real constexpr operator"" _f(unsigned long long v) {
+real constexpr operator""_f(unsigned long long v) {
   return real(v);
 }
 
-float32 constexpr operator"" _f32(long double v) {
+float32 constexpr operator""_f32(long double v) {
   return float32(v);
 }
-float32 constexpr operator"" _f32(unsigned long long v) {
+float32 constexpr operator""_f32(unsigned long long v) {
   return float32(v);
 }
 
-float32 constexpr operator"" _fs(long double v) {
+float32 constexpr operator""_fs(long double v) {
   return float32(v);
 }
-float32 constexpr operator"" _fs(unsigned long long v) {
+float32 constexpr operator""_fs(unsigned long long v) {
   return float32(v);
 }
 
-float64 constexpr operator"" _f64(long double v) {
+float64 constexpr operator""_f64(long double v) {
   return float64(v);
 }
-float64 constexpr operator"" _f64(unsigned long long v) {
+float64 constexpr operator""_f64(unsigned long long v) {
   return float64(v);
 }
 
-float64 constexpr operator"" _fd(long double v) {
+float64 constexpr operator""_fd(long double v) {
   return float64(v);
 }
-float64 constexpr operator"" _fd(unsigned long long v) {
+float64 constexpr operator""_fd(unsigned long long v) {
   return float64(v);
 }
 
diff --git a/taichi/common/types.h b/taichi/common/types.h
index b4728a6..fd369e2 100644
--- a/taichi/common/types.h
+++ b/taichi/common/types.h
@@ -38,38 +38,38 @@ using real = float32;
 
 // Float literal for both float32/64
 // (Learned from https://github.com/hi2p-perim/lightmetrica-v2)
-real constexpr operator"" _f(long double v) {
+real constexpr operator""_f(long double v) {
   return real(v);
 }
-real constexpr operator"" _f(unsigned long long v) {
+real constexpr operator""_f(unsigned long long v) {
   return real(v);
 }
 
-float32 constexpr operator"" _f32(long double v) {
+float32 constexpr operator""_f32(long double v) {
   return float32(v);
 }
-float32 constexpr operator"" _f32(unsigned long long v) {
+float32 constexpr operator""_f32(unsigned long long v) {
   return float32(v);
 }
 
-float32 constexpr operator"" _fs(long double v) {
+float32 constexpr operator""_fs(long double v) {
   return float32(v);
 }
-float32 constexpr operator"" _fs(unsigned long long v) {
+float32 constexpr operator""_fs(unsigned long long v) {
   return float32(v);
 }
 
-float64 constexpr operator"" _f64(long double v) {
+float64 constexpr operator""_f64(long double v) {
   return float64(v);
 }
-float64 constexpr operator"" _f64(unsigned long long v) {
+float64 constexpr operator""_f64(unsigned long long v) {
   return float64(v);
 }
 
-float64 constexpr operator"" _fd(long double v) {
+float64 constexpr operator""_fd(long double v) {
   return float64(v);
 }
-float64 constexpr operator"" _fd(unsigned long long v) {
+float64 constexpr operator""_fd(unsigned long long v) {
   return float64(v);
 }
 
